Source/Core/loadJsonp.js

```diff
@@ -0,0 +1,118 @@
+/*global define*/
+define([
+    '../ThirdParty/Uri',
+    '../ThirdParty/when',
+    './combine',
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './objectToQuery',
+    './queryToObject'
+], function(
+    Uri,
+    when,
+    combine,
+    defaultValue,
+    defined,
+    DeveloperError,
+    objectToQuery,
+    queryToObject) {
+    "use strict";
+
+    /**
+     * Requests a resource using JSONP.
+     *
+     * @exports loadJsonp
+     *
+     * @param {String} url The URL to request.
+     * @param {Object} [options] Object with the following properties:
+     * @param {Object} [options.parameters] Any extra query parameters to append to the URL.
+     * @param {String} [options.callbackParameterName='callback'] The callback parameter name that the server expects.
+     * @param {Proxy} [options.proxy] A proxy to use for the request. This object is expected to have a getURL function which returns the proxied URL, if needed.
+     * @returns {Promise.<Object>} a promise that will resolve to the requested data when loaded.
+     *
+     * @see {@link http://wiki.commonjs.org/wiki/Promises/A|CommonJS Promises/A}
+     *
+     * @example
+     * // load a data asynchronously
+     * Cesium.loadJsonp('some/webservice').then(function(data) {
+     *     // use the loaded data
+     * }).otherwise(function(error) {
+     *     // an error occurred
+     * });
+     */
+    var loadJsonp = function(url, options) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(url)) {
+            throw new DeveloperError('url is required.');
+        }
+        //>>includeEnd('debug');
+
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        //generate a unique function name
+        var functionName;
+        do {
+            functionName = 'loadJsonp' + Math.random().toString().substring(2, 8);
+        } while (defined(window[functionName]));
+
+        var deferred = when.defer();
+
+        //assign a function with that name in the global scope
+        window[functionName] = function(data) {
+            deferred.resolve(data);
+
+            try {
+                delete window[functionName];
+            } catch (e) {
```

Comment:
I searched around to refresh my memory and yes, `delete window.foo` always throws on old IE.  http://stackoverflow.com/questions/1073414/deleting-a-window-property-in-ie  

So I'd suggest keeping it, since it doesn't seem too burdensome to do so.